Spigot-Server-Patches/0565-Add-additional-open-container-api-to-HumanEntity.patch

```diff
@@ -0,0 +1,127 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: JRoy <joshroy126@gmail.com>
+Date: Wed, 26 Aug 2020 02:12:31 -0400
+Subject: [PATCH] Add additional open container api to HumanEntity
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+index 98768b5df93fd1bf94858149e15a8c9a16439695..02d52c2a9c95ca60f08720a44b825c75e0ba8513 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftHumanEntity.java
+@@ -450,6 +450,116 @@ public class CraftHumanEntity extends CraftLivingEntity implements HumanEntity {
+         return this.getHandle().activeContainer.getBukkitView();
+     }
+ 
++    // Paper start - Add additional containers
++    @Override
++    public InventoryView openAnvil(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
++            if (block.getType() != Material.ANVIL) {
++                return null;
++            }
++        }
++        if (location == null) {
++            location = getLocation();
++        }
++        getHandle().openContainer(((net.minecraft.server.BlockAnvil) Blocks.ANVIL).getInventory(null, getHandle().world, new BlockPosition(location.getBlockX(), location.getBlockY(), location.getBlockZ())));
++        if (force) {
++            getHandle().activeContainer.checkReachable = false;
++        }
++        return getHandle().activeContainer.getBukkitView();
++    }
++
++    @Override
++    public InventoryView openCartographyTable(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
++            if (block.getType() != Material.CARTOGRAPHY_TABLE) {
++                return null;
++            }
++        }
++        if (location == null) {
++            location = getLocation();
++        }
++        getHandle().openContainer(((net.minecraft.server.BlockCartographyTable) Blocks.CARTOGRAPHY_TABLE).getInventory(null, getHandle().world, new BlockPosition(location.getBlockX(), location.getBlockY(), location.getBlockZ())));
++        if (force) {
++            getHandle().activeContainer.checkReachable = false;
++        }
++        return getHandle().activeContainer.getBukkitView();
++    }
++
++    @Override
++    public InventoryView openGrindstone(Location location, boolean force) {
++        if (!force) {
++            Block block = location.getBlock();
```

Comment:
You check if`location` is `null` below, but not here - is this intentional? If so, why?